app/components/PublishDialog.tsx

```diff
@@ -0,0 +1,101 @@
+import { Button } from '@/components/ui/button'
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from '@/components/ui/dialog'
+import { Input } from '@/components/ui/input'
+import { Label } from '@/components/ui/label'
+import { useElectric } from 'app/Providers'
+import { createStrategy } from 'app/actions/createStrategy'
+import { useRouter } from 'next/navigation'
+import { useEffect, useState } from 'react'
+import slugify from 'slugify'
+import { v4 as uuid } from 'uuid'
+
+interface Props {
+  query: string
+}
+
+export const PublishDialog = ({ query }: Props) => {
+  const electric = useElectric()
+  const router = useRouter()
+  const [isOpen, setIsOpen] = useState(false)
+  const [id, setId] = useState(uuid())
+  const [name, setName] = useState(id)
+  const [description, setDescription] = useState('')
+
+  useEffect(() => {
+    setId(slugify(name, { lower: true }))
+  }, [name])
+
+  if (!electric) {
+    return null
```

Comment:
Don't show publish, if Electric is not ready! (or Postgres is not ready, which makes Electric service 'not ready')